pallets/bridge_transfer/src/lib.rs

```diff
@@ -196,9 +221,20 @@ impl<T: Config> Module<T> {
 		let mut encoded_token_id: Vec<u8> = token_id;
 		encoded_token_id.reverse();
 
-		let metadata: Vec<u8> = [encoded_round_id, encoded_token_id, encoded_payload_len, payload.clone()].concat();
-
-		Self::deposit_event(Event::BTCSignedTxSend(round_id, dest_id, resource_id, payload, sequence));
+		let metadata: Vec<u8> = [
+			encoded_round_id,
+			encoded_token_id,
+			encoded_payload_len,
+			payload.clone(),
+		]
+		.concat();
```

Comment:
Let's use the encoded structure directly. Not necessary to decode round_id / token_id / payload_len, etc

```
let metadata = payload.encode();
```